examples/features/gracefulstop/client/main.go

```diff
@@ -0,0 +1,67 @@
+/*
+ *
+ * Copyright 2024 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Binary client demonstrates sending multiple requests to server and observe
+// graceful stop.
```

Comment:
Is there some way this can actually prove that the stop was graceful?  E.g. by starting a stream that runs until requests start failing, and then closing the stream from the client and having it exit successfully by getting a response from the server at that time?

E.g. the server could respond with a message indicating how many requests it processed.

Reply:
So, here is what i did now

I wrote an example server which has a server streaming handler that sends 5 messages to client with 1 second delay in-between. After 5 messages, it sends an error to client. If we interrupt server during those 5 messages, server will make sure to complete those 5 messages followed by the error which demonstrates graceful shutdown.

Reply:
^C to cancel makes this hard to test.  We _should_ be covering this (and all examples) in https://github.com/grpc/grpc-go/blob/master/examples/examples_test.sh.  Please find a way to demonstrate this but to be testable and ideally not require more than a couple seconds to test.  I think having the client do `CloseSend` after unary requests start failing seems pretty nice, and counting the RPCs is something that can easily be done on both sides to show that things are still working on this connection even after the server has gracefully stopped.  If you have other ideas, that's fine too, but the example should be tested.

Reply:
removed `ctrl+c` requirement. Here is what i am doing now.
- Server will spawn a go routine that waits for start stream signal before initiating graceful stop. 
- When the client starts server streaming, handler will signal the server to initiate graceful stop and starts sending messages to client
- Client will receives 5 messages and then close the stream. This will trigger handler to finish the rpc.
- Graceful stop completes then

Reply:
As discussed offline, i have updated the approach to the combination of client streaming and unary rpc. So, client will start client streaming which will initiates graceful shutdown after 1 sec. During this 1 sec, client will send as many unary requests it can and will close the stream once graceful shutdown is initiated.